### PR TITLE
[MBO-1249] Catch empty employee email exception

### PR DESCRIPTION
**Pull request by sowbiba**
Fix for #620 

The exception won't block the whole backoffice anymore. You can use MBO in degraded mode and have a warning in the marketplace page
Actions on modules won't work until the module is resetted

### CODE REVIEW

File: src/Api/Security/AdminAuthenticationProvider.php [hunk 1]
```diff
@@ -37,6 +38,8 @@
 
 class AdminAuthenticationProvider
 {
+    private const DEFAULT_USER_ID = 42;
```
Comment (sowbiba) [inline src/Api/Security/AdminAuthenticationProvider.php:41]: https://github.com/search?q=repo%3APrestaShop%2FPrestaShop%20Employee(42)&type=code

File: src/Api/Security/AdminAuthenticationProvider.php [hunk 2]
```diff
@@ -76,8 +79,11 @@ public function __construct(
         $this->cacheProvider = $cacheProvider;
     }
 
-    public function createApiUser(): Employee
+    public function createApiUser(): ?Employee
```
Comment (sowbiba) [inline src/Api/Security/AdminAuthenticationProvider.php:82]: https://github.com/search?q=repo%3APrestaShop%2FPrestaShop%20Employee(42)&type=code

File: src/Api/Security/AdminAuthenticationProvider.php [hunk 3]
```diff
@@ -234,4 +267,11 @@ private function getJwtTokenCacheKey(): string
     {
         return sprintf('mbo_jwt_token_%s', Config::getShopMboUuid());
     }
+
+    private function getDefaultUserToken()
```
Comment (intraordinaire) [inline src/Api/Security/AdminAuthenticationProvider.php:271]:
```suggestion
    private function getDefaultUserToken(): string
```

File: src/Traits/Hooks/UseDisplayDashboardTop.php [hunk 4]
```diff
@@ -151,6 +162,15 @@ protected function displayPushOnConfigurationPage(string $moduleName): string
         return $this->fetch('module:ps_mbo/views/templates/hook/push-configuration.tpl');
     }
 
+    private function displayFailedApiUser()
+    {
+        $this->smarty->assign([
+            'explanation' => $this->trans('The creation of the MBO Admin user failed. Your marketplace will work in degraded mode. Some actions on modules won\'t be available. Please try to reset the module or contract an admin for help', [], 'Modules.Mbo.Global'),
```
Comment (intraordinaire) [inline src/Traits/Hooks/UseDisplayDashboardTop.php:168]:
```suggestion
            'explanation' => $this->trans('The creation of the MBO Admin user failed. Your marketplace will work in degraded mode. Some actions on modules won\'t be available. Please try to reset the module or contact an admin for help', [], 'Modules.Mbo.Global'),
```

Comment (intraordinaire) [reply src/Traits/Hooks/UseDisplayDashboardTop.php:168]: We have to validate the wording here anyway.

File: src/Traits/Hooks/UseActionAdminControllerSetMedia.php [hunk 5]
```diff
@@ -46,8 +46,9 @@ public function hookActionAdminControllerSetMedia(): void
     {
         if (Tools::getValue('controller') === "AdminPsMboModule") {
             $this->context->controller->addJs($this->getPathUri() . 'views/js/upload_module_with_cdc.js?v=' . $this->version);
+            $this->context->controller->addJs($this->getPathUri() . 'views/js/failed-api-user.js?v=' . $this->version);
```
Comment (intraordinaire) [inline src/Traits/Hooks/UseActionAdminControllerSetMedia.php:49]: I just see this, bur there is no file like this, no?

**Review by intraordinaire — APPROVED** [review]

**Pull request merged**